tools/cert-sidecar/app/CertificatesMgmt.js

```diff
@@ -0,0 +1,289 @@
+    /**
+     *  Path to  crl
+     * @type {String}
+     */
+    this.pathCrl = filenameInCertFolder(configCerts['files.crl']);
+    /**
+     *  Path to ca
+     * @type {String}
+     */
+    this.pathCA = filenameInCertFolder(configCerts['files.ca']);
+    /**
+     *  Path to certificate
+     * @type {String}
+     */
+    this.pathCert = filenameInCertFolder(configCerts['files.cert']);
+    /**
+     *  Path to privatekey
+     * @type {String}
+     */
+    this.pathKey = filenameInCertFolder(configCerts['files.key']);
+  }
+
+  /**
+   * Initialize calling to create files
+   * And calling the crons
+   */
+  async init() {
+    logger.info('Init: Initializing...');
+    try {
+      // create dir if not exist
+      await createDir(configCerts['files.basepath']);
+
+      logger.debug('Init: Calling generatePrivateKey...');
+      await this.generatePrivateKey();
+      logger.debug('Init: ...generatePrivateKey called.');
+
+      logger.debug('Init: Calling generateCertificate...');
+      await this.generateCertificate();
+      logger.debug('Init: ...generateCertificate called.');
+
+      logger.debug('Init: Calling retrieveCaCert...');
+      await this.retrieveCaCert();
+      logger.debug('Init: ...retrieveCaCert called.');
+
+      logger.debug('Init: Calling retrieveCRL...');
+      if (configCerts.crl) {
+        await this.retrieveCRL();
+        logger.debug('Init: ...retrieveCRL called.');
+      } else {
+        logger.info('Init: CRL is disabled');
+      }
+    } catch (e) {
+      logger.error('init:', e);
+      throw e;
+    }
+    logger.info('Init: ...initialized');
+  }
+
+  /**
+   * Check if certificate and ca certificate will expire
+   * and get a new one if is the case
+   */
+  async certsWillExpire() {
+    logger.info('certsWillExpire: Checking if broker\'s certificates are expired...');
+    try {
+      if (await X509Utils.isCertExpiredInSec(
+        this.ca,
+        configCerts['expiration.checkend'],
+      )) {
+        logger.info('certsWillExpire: Getting new ca certificate');
+        await deleteFile(this.pathCA);
+        this.ca = null;
+        await this.retrieveCaCert();
+      } else {
+        logger.info('certsWillExpire: ca certificate ok ');
+      }
+
+      if (await X509Utils.isCertExpiredInSec(
+        this.cert,
+        configCerts['expiration.checkend'],
+      )) {
+        logger.info('certsWillExpire: Getting new certificate');
+        await deleteFile(this.pathKey);
+        await deleteFile(this.pathCert);
+        this.privateKey = null;
+        this.cert = null;
+        await this.generateCertificate();
+      } else {
+        logger.info('certsWillExpire: certificate ok ');
+      }
+    } catch (e) {
+      logger.error('certsWillExpire:', e);
+      throw e;
+    }
+    logger.info('certsWillExpire: ...end check if broker\'s certificates are expired.');
+  }
+
+  /**
+   * Check if certificate has revoke
+   * and get a new one if is the case
```

Comment:
My suggestion for this phrase:
Checks whether the certificate has been revoked and retrieves a new one if it is the case.